### PR TITLE
chore: update afya_moja_core package and validation in nickname input

### PR DESCRIPTION
**Pull request by LUCYOGUTU**

### CODE REVIEW

**Comment by LUCYOGUTU** [comment]
WhatsApp Image 2022-10-14 at 09 11 58
WhatsApp Image 2022-10-14 at 09 12 42

**Review by EugeneDevv — APPROVED** [review]

File: lib/domain/core/value_objects/app_strings.dart [hunk 1]
```diff
@@ -380,6 +380,7 @@ String getEnrollmentDuration({
 const String congratulationsPageDescription =
     'Please create a nickname (Visible to community members)';
 const String nameInputValidateString = 'Kindly input a nickname';
+const String errorNicknameInputString = 'Allowed inputs are numbers, characters and underscores';
```
Comment (abiudrn) [inline lib/domain/core/value_objects/app_strings.dart:383]: Change this to 
```suggestion
const String errorNicknameInputString = 'Please enter letters, numbers or underscores only';
```

File: lib/presentation/onboarding/set_nickname/pages/set_nickname_page.dart [hunk 2]
```diff
@@ -35,6 +34,7 @@ class _SetNickNamePageState extends State<SetNickNamePage> {
 
   @override
   Widget build(BuildContext context) {
+    final RegExp inputNicknameRegex = RegExp(r'^[0-9a-zA-Z_]{1,30}$');
```
Comment (abiudrn) [inline lib/presentation/onboarding/set_nickname/pages/set_nickname_page.dart:37]: Extract this into a well named constant

**Review by abiudrn — CHANGES_REQUESTED** [review]

File: lib/domain/core/value_objects/constants.dart [hunk 3]
```diff
@@ -0,0 +1 @@
+RegExp nameRegexString = RegExp(r'^[0-9a-zA-Z_]{1,30}$');
```
Comment (abiudrn) [inline lib/domain/core/value_objects/constants.dart:1]:
```suggestion
RegExp userNameRegex = RegExp(r'^[0-9a-zA-Z_]{1,30}$');
```

**Review by abiudrn — CHANGES_REQUESTED** [review]

**Review by abiudrn — APPROVED** [review]

**Review by abiudrn — APPROVED** [review]

**Pull request merged**